src/draive/conversation/realtime/types.py

```
@@ -1,25 +1,111 @@
+        return await ctx.state(cls).reading()
+
+    @classmethod
+    async def reader(cls) -> AsyncIterator[ConversationStreamElement]:
+        session: Self = ctx.state(cls)
+        while True:  # breaks on exception
+            yield await session.reading()
+
+    @classmethod
+    async def write(
+        cls,
+        input: ConversationStreamElement,  # noqa: A002
```

Pylint (3.3.7)

[warning] Redefining built-in 'input'

(W0622)